Szymon_zadanie10/src/test/java/Szymon_zadanie10/page/common/BaseStorePage.java

```diff
@@ -0,0 +1,121 @@
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.List;
+import java.util.Objects;
+
+public class BaseStorePage extends Page {
+    private HeaderComponent header;
+    private ProductOptionsComponent productOptionsComponent;
+    private AddToCartPopUpComponent addToCartPopUpComponent;
+    private ContextMenuComponent contextMenuComponent;
+    @FindBy(xpath = "//span[@title='Continue shopping']//span")
+    private WebElement continueShoppingButton;
+    @FindBy(css = ".ajax_cart_quantity")
+    private WebElement textNumberOfItemsInCartFromPopup;
+    @FindBy (xpath = "//a[@title='Proceed to checkout']//span")
+    private WebElement proceedToCheckout;
+
+    public BaseStorePage(WebDriver driver) {
+        super(driver);
+        header = new HeaderComponent(driver);
+        productOptionsComponent = new ProductOptionsComponent(driver);
+        addToCartPopUpComponent = new AddToCartPopUpComponent(driver);
+        contextMenuComponent = new ContextMenuComponent(driver);
+    }
+
+    public HeaderComponent getHeaderComponent() {
+        return header;
+    }
+
+    public ProductOptionsComponent getProductOptionsComponent() {
+        return productOptionsComponent;
+    }
+
+    public AddToCartPopUpComponent getAddToCartPopUpComponent() {
+        return addToCartPopUpComponent;
+    }
+
+    public ContextMenuComponent getContextMenuComponent() {
+        return contextMenuComponent;
+    }
+
+    public ProductBlousePage getProductBlousePage() {
+        return new ProductBlousePage(driver);
+    }
+
+    public BaseStorePage continueShopping() {
+        //Waiting for the popup to display
+        //        waitUntil(p-> continueShoppingButton.isDisplayed()); < co tutaj trzeba zrobić żeby była dostępna metoda 'elementToBeClickable'
+        WebDriverWait wait = new WebDriverWait(driver, 5);
+        WebElement element = wait.until(ExpectedConditions.elementToBeClickable(continueShoppingButton));
+        continueShoppingButton.click();
+        waitUntil(p -> !continueShoppingButton.isDisplayed());
+        return new BaseStorePage(driver);
+    }
+
+    public WebElement getContinueShoppingButton() {
+        return continueShoppingButton;
+    }
+
+    public WebElement findElementByText(List<WebElement> webElements, String text) {
+        return webElements
+                .stream()
+                .filter(webElement -> Objects.equals(webElement.getText(), text))
+                .findFirst()
+                .orElseThrow(() -> new NoSuchElementException("No WebElement found containing " + text));
+    }
+
+    public BaseStorePage addToCart(String itemTitle) {
+        String xpathForItemAddButton = "//div[@itemscope][.//a[@title='" + itemTitle + "']]//a/span[contains(text(),'Add to cart')]";
```

Comment:
Tu sie wkradlo troche balaganu:
Xpathy, jesli sa prametryzowane, to zrobmy z nich stale:
```
private final static String ADD_BUTTON_XPATH = "//div[@itemscope][.//a[@title='%s']]//a/span[contains(text(),'Add to cart')]";

String.format(ADD_BUTTON_XPATH, itemTitle);
```
etc.

uzyj metody waitUntil zamiast nowych WebDriverWait.

Poza tym spoko.

Reply:
Tworzyłem nowe WebDriverWait bo  waitUntil dawało mi tylko 3 opcje waita

Reply:
```waitUntil(ExpectedConditions.elementToBeClickable(By.xpath(xpathForItem)));```
To powinno normalnie dzialac...

Reply:
hmm ale z tymi stałymi xpathami to musisz wytłumaczyć 1. dlaczego final static 2.robimy zmienną %s której intellij nie linkuje tylko żeby wyciągnąć do ewentualnych innych metod?